flink-runtime/src/test/java/org/apache/flink/runtime/source/coordinator/SourceCoordinatorContextTest.java

```diff
@@ -155,6 +156,40 @@ private void testAssignSplitToUnregisterdReader(boolean fromCoordinatorExecutor)
                 "Cannot assign splits");
     }
 
+    @Test
+    public void testExceptionInRunnableFailsTheJob()
+            throws InterruptedException, ExecutionException {
+        ManuallyTriggeredScheduledExecutorService manualWorkerExecutor =
+                new ManuallyTriggeredScheduledExecutorService();
+        // need the factory to have the exception handler set
+        final ManuallyTriggeredScheduledExecutorService coordinatorExecutorWithExceptionHandler =
+                new ManuallyTriggeredScheduledExecutorService(1, coordinatorThreadFactory);
```

Comment:
This implementation executes `Runnables` directly in the thread that has executed the `trigger...` method, so the exception handler is basically ignored.

I guess this might have been the reason for introducing the `wrap` method on the context, otherwise this test shouldn't pass.

Also the changes to `ManuallyTriggeredScheduledExecutorService` don't seem to have any effect, so we should be able to get rid of them.

Can we do something as simple as ... ?

```java
        final ScheduledExecutorService executorService =
                Executors.newSingleThreadScheduledExecutor(
                        new SourceCoordinatorProvider.CoordinatorExecutorThreadFactory("test", operatorCoordinatorContext));
```

Another option would be adding support for the exception handler to `ManuallyTriggeredScheduledExecutorService` (basically wrapping `next.run();` in the `trigger` method with `try/catch`.

Reply:
see my previous comment above it is only for exceptions in the threadpool itself.

Reply:
another comment, `wrap()` was introduced in the PR before using `ManuallyTriggeredScheduledExecutorService` when `Executors.newSingleThreadScheduledExecutor` was in use. It was introduced because `ScheduledFutureTask` catches its own exceptions.